### ISSUE TITLE
BCLib 0.4.3 crashes fabric loader 0.11.7 server when client connects

### ISSUE THREAD

**Issue by gniftygnome** [paulevsGitch/BCLib#36]
It looks like BCLib 0.4.3 is not compatible with fabric loader 0.11.7.  It works fine on the client (including integrated server) and it works on the server until somebody connects.  Then the server crashes.  Downgrading to BCLib 0.4.2 resolves the issue.

Top of stack:
```
java.lang.NoClassDefFoundError: net/fabricmc/loader/impl/metadata/ModMetadataParser
        at Not Enough Crashes deobfuscated stack trace.(1.17.1+build.63)
        at ru.bclib.util.ModUtil.lambda$getMods$0(ModUtil.java:60)
        at ru.bclib.util.PathUtil.fileWalker(PathUtil.java:73)
        at ru.bclib.util.ModUtil.getMods(ModUtil.java:51)
        at ru.bclib.api.dataexchange.handler.autosync.HelloClient.serializeDataOnServer(HelloClient.java:86)
        at ru.bclib.api.dataexchange.DataHandler$FromServer.sendToClient(DataHandler.java:262)
        at ru.bclib.api.dataexchange.BaseDataHandler.reply(BaseDataHandler.java:47)
        at ru.bclib.api.dataexchange.handler.autosync.HelloServer.runOnServerGameThread(HelloServer.java:106)
        at ru.bclib.api.dataexchange.DataHandler$FromClient.lambda$receiveFromClient$0(DataHandler.java:165)
```

Full crash report:  [crash-2021-11-03_15.23.15-server.txt](https://github.com/paulevsGitch/BCLib/files/7471101/crash-2021-11-03_15.23.15-server.txt)

**Comment by quiqueck** [comment]
What Version of Fabric API are you using?

**Comment by gniftygnome** [comment]
Fabric API 0.41.3+1.17

**Comment by quiqueck** [comment]
For Context: BCLib 0.4.2 was built against the 0.11.x Loader. 0.12.x appears to have changed some classes we rely on (see #34) which caused the hotfix release for 0.4.3 which is built for 0.12.x loaders

**Comment by gniftygnome** [comment]
Yeah, I saw that ... I have actually tried 0.12.x a few times now but there are multiple mods we are using that have issues with it (formerly including Better End).  Then you have the new mods built on the current version of the example mod which requires 0.12.x by default.  hah

**Comment by quiqueck** [comment]
Yeah, the next hotfix (0.4.4) should work with both loaders again.

**Comment by quiqueck** [comment]
Fixed in 0.4.4. If you are stille having problems, please repoen the issue.
